crates/vim/src/surrounds.rs

```diff
@@ -241,21 +241,15 @@ impl Vim {
                         },
                     };
 
-                    // Determines whether space should be added/removed after
+                    // Determines whether space should be added after
                     // and before the surround pairs.
-                    // For example, using `cs{[` will add a space before and
-                    // after the pair, while using `cs{]` will not, notice the
-                    // use of the closing bracket instead of the opening bracket
-                    // on the target object.
-                    // In the case of quotes, the opening and closing is the
-                    // same, so no space will ever be added or removed.
-                    let surround = match target {
-                        Object::Quotes
-                        | Object::BackQuotes
-                        | Object::AnyQuotes
-                        | Object::MiniQuotes
-                        | Object::DoubleQuotes => true,
-                        _ => pair.end != surround_alias((*text).as_ref()),
+                    // Space is only added in the following cases:
+                    // - new surround is not quote and is opening bracket (({[<)
+                    // - new surround is quote and original was also quote
```

Comment:
Is this comment correct? If the original surround is a quote (i.e. `'`) and the new surround is also a quote (i.e. `"`) it should _not_ add space, right?

Reply:
I was struggling to write a comment that clearly explained how the `surround` variable impacts the result when set to `true`. 

When it's set to `true` it'll either add a new space:

`'a'` → `cs'{` → `{ a }`

Or it'll preserve the existing space:

`' a '` → `cs'"` → `" a "`

So I guess a better comment would clarify that "Space is only added or preserved...". I'll be updating the change surround commands to more closely match [tpope/vim-surround](https://github.com/tpope/vim-surround), as discussed [here](https://github.com/zed-industries/zed/issues/38169#issuecomment-3304129461), and will review this comment as part of that work. Thanks for pointing it out! 🙇